api/web3server_utils.go

```diff
@@ -117,7 +118,19 @@ func (svr *web3Handler) getTransactionFromActionInfo(blkHash hash.Hash256, selp
 	if err != nil {
 		return nil, err
 	}
-	tx, err := action.RawTxToSignedTx(ethTx, svr.coreService.EVMNetworkID(), selp.Signature())
+	var (
+		signer   types.Signer
+		encoding = iotextypes.Encoding(selp.Encoding())
+	)
+	if encoding != iotextypes.Encoding_IOTEX_PROTOBUF {
+		signer, err = action.NewEthSigner(encoding, svr.coreService.EVMNetworkID())
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		signer = types.HomesteadSigner{}
```

Comment:
when will this happen?

Reply:
this is for native tx (the signature format is same as Homestead)
I reverted as Haixiang mentioned, this part should be next PR